R/edit_attribute.R

```diff
@@ -0,0 +1,77 @@
+#' @param domain The new domain to give to the attribute
+#' @param measurementScale The new measurementScale to give to the attribute
+#' @param unit The new unit to give to the attribute
+#' @param numberType The new numberType to give to the attribute
+#' @param definition The new definition (for textDomain) to give to the attribute
+#' @param formatString The new formatString to give to the attribute
+#' @param missingValueCode The new missing value code to give to the attribute
+#' @param missingValueCodeExplanation The new missing value code explaination to give to the attribute
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' #To change an attribute's name and add a missing value code where there hadn't been one:
+#' new_eml <- edit_attribute(eml, 2, 8, attributeName = "new name", missingValueCode = "NA", missingValueCodeExplanation = "Data unavailable")
+#' #To change an attribute's attributeDefinition, as well as change measurementScale from nominal to ratio (requires also changing domain, unit, and numberType, as well as setting definition to NA):
+#' new_eml <- edit_attribute(eml, 1, 2, attributeDefinition = "new definition", domain = "numericDomain",
+#' measurementScale = "ratio", unit = "dimensionless", numberType = "whole", definition = NA)
+#' }
+#'
+#'
+edit_attribute <- function(eml, dataTableNumber, attributeNumber, attributeName = NULL, attributeDefinition = NULL, domain=NULL,
+                           measurementScale = NULL, unit = NULL, numberType = NULL, definition = NULL, formatString = NULL,
+                           missingValueCode = NULL, missingValueCodeExplanation = NULL){
+
+    data<-EML::get_attributes(eml@dataset@dataTable[[dataTableNumber]]@attributeList)
+    attributeTable<-data.frame(data$attributes) #this excludes the factor table from enumerated domain.
+
+    if(!is.null(attributeName)==TRUE){
+        attributeTable[attributeNumber,1] <- attributeName
+    }
+    if(!is.null(attributeDefinition)==TRUE){
+        attributeTable[attributeNumber,17] <- attributeDefinition
+    }
+    if(!is.null(measurementScale)==TRUE){
+        attributeTable[attributeNumber,16] <- measurementScale
+    }
+    if(!is.null(domain)==TRUE){
+        attributeTable[attributeNumber,2] <- domain
+    }
+    if(!is.null(unit)==TRUE){
+        attributeTable[attributeNumber,6] <- unit
+    }
+    if(!is.null(numberType)==TRUE){
+        attributeTable[attributeNumber,7] <- numberType
+    }
+    if(!is.null(definition)==TRUE){
+        attributeTable[attributeNumber,9] <- definition
+    }
+    if(!is.null(formatString)==TRUE){
+        attributeTable[attributeNumber,8] <- formatString
+    }
+    if(!is.null(missingValueCode)==TRUE){
+        attributeTable[attributeNumber,14] <- missingValueCode
+    }
+    if(!is.null(missingValueCodeExplanation)==TRUE){
+        attributeTable[attributeNumber,15] <- missingValueCodeExplanation
+    }
+
+    attribute.list<-EML::set_attributes(attributeTable, factors = data$factors)
```

Comment:
for our package, we're trying to stick with underscores unless we're referring to EML/etc. Unless you have a reason for keeping it, I'd suggest changing `attribute.list` to `attribute_list`